app/src/main/java/app/revanced/manager/ui/viewmodel/PatcherViewModel.kt

```diff
@@ -243,18 +288,54 @@ class PatcherViewModel(
     fun install(installType: InstallType) = viewModelScope.launch {
         try {
             isInstalling = true
+
+            // If the app is currently installed
+            val existingPackageInfo = pm.getPackageInfo(packageName)
+            if (existingPackageInfo != null) {
+                // Check if the app version is less than the installed version
+                if (pm.versionNameToInt(input.selectedApp.version) < pm.versionNameToInt(existingPackageInfo.versionName)) {
+                    // Exit if the selected app version is less than the installed version
+                    installerStatusDialogModel.packageInstallerStatus = PackageInstaller.STATUS_FAILURE_CONFLICT
+                    return@launch
+                }
+            }
```

Comment:
That's not what I meant in my review, use versionCodes, not versionName

versionName doesn't determine if it can be installed or not, versionCode does, if its equal or bigger then it can be installed

Reply:
I have now switched it to use:
```kotlin
fun getVersionCode(packageInfo: PackageInfo): Int {
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
        packageInfo.longVersionCode.toInt()
    } else {
        packageInfo.versionCode
    }
}
```
Is this correct?

Reply:
Make sure to check if its equal too, because equal can be installed